plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanel.java

```diff
@@ -44,4 +58,61 @@ public boolean savePreferences() {
     return false;
   }
 
+  private void createConfigurationFilesSection() {
+    useValuesButton = new Button(this, SWT.CHECK);
+    useValuesButton.setText(Messages.getString("use.config.values"));
+    useValuesButton.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false, 3, 1));
+
+    gaeConfigFolderLabel = new Label(this, SWT.LEFT);
+    gaeConfigFolderLabel.setText(Messages.getString("config.folder.location"));
+    gaeConfigFolderText = new Text(this, SWT.LEFT | SWT.SINGLE | SWT.BORDER);
+    gaeConfigFolderBrowseButton = new Button(this, SWT.PUSH);
+    gaeConfigFolderBrowseButton.setText(Messages.getString("browse.button"));
+    gaeConfigFolderBrowseButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+      public void widgetSelected(SelectionEvent event) {
+              browseForConfigFolder();
+            }
+        });
+
+    dockerFileLabel = new Label(this, SWT.LEFT);
+    dockerFileLabel.setText(Messages.getString("docker.file.location"));
+    dockerFileText = new Text(this, SWT.LEFT | SWT.SINGLE | SWT.BORDER);
+    dockerFileBrowseButton = new Button(this, SWT.PUSH);
+    dockerFileBrowseButton.setText(Messages.getString("browse.button"));
+    dockerFileBrowseButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+      public void widgetSelected(SelectionEvent event) {
+              browseForDockerFile();
+            }
+        });
+
+    GridLayoutFactory.fillDefaults().numColumns(3).generateLayout(this);
+  }
+
+  private void browseForConfigFolder() {
+    String last = gaeConfigFolderText.getText().trim();
+    DirectoryDialog dialog = new DirectoryDialog(getShell(), SWT.SINGLE);
+    dialog.setText(Messages.getString("config.folder.browse.button.text"));
+    dialog.setMessage(Messages.getString("config.folder.browse.button.message"));
+    dialog.setFilterPath(last);
+    String result = dialog.open();
```

Comment:
You also need some way to convey validation status so the OK button is disabled with error text…